pydantic/schema.py

```diff
@@ -849,32 +849,65 @@ def encode_default(dft: Any) -> Any:
 
 
 _map_types_constraint: Dict[Any, Callable[..., type]] = {int: conint, float: confloat, Decimal: condecimal}
-_field_constraints = {
-    'min_length',
-    'max_length',
-    'regex',
-    'gt',
-    'lt',
-    'ge',
-    'le',
-    'multiple_of',
-    'min_items',
-    'max_items',
+_field_constraints = {  # field constraints with the default value
+    ('min_length', None),
+    ('max_length', None),
+    ('regex', None),
+    ('gt', None),
+    ('lt', None),
+    ('ge', None),
+    ('le', None),
+    ('multiple_of', None),
+    ('min_items', None),
+    ('max_items', None),
+    ('allow_mutation', True),
 }
 
 
-def get_annotation_from_field_info(annotation: Any, field_info: FieldInfo, field_name: str) -> Type[Any]:  # noqa: C901
+def get_constraints(field_info: FieldInfo) -> Set[str]:
+    """Gets the constraints set on the field by comparing the constraint value with it's default value
+
+    :param field_info: an instance of FieldInfo
+    :return: the constraints set on field_info
     """
-    Get an annotation with validation implemented for numbers and strings based on the field_info.
+    return {
+        f[0] for f in _field_constraints if getattr(field_info, f[0]) is not None and getattr(field_info, f[0]) != f[1]
+    }
+
 
+def get_annotation_from_field_info(
+    annotation: Any, field_info: FieldInfo, field_name: str, validate_assignment: bool = False
```

Comment:
I had to add a default value for validate_assignment for the fastAPI check to pass as fastApi uses this function.  Should fastAPI be updated to include this parameter?  I haven't looked at the fastAPI codebase yet.